Taskfile.yml

```diff
@@ -63,21 +63,56 @@ tasks:
     desc: Run the unit tests.
     cmds:
       # One day I will understand how to use -coverpkg=./... :-(
-      - go test -count=1 -short -coverprofile=coverage.out ./...
+      # FIXME - go test -count=1 -short -coverprofile=coverage.out ./...
+      - go test -count=1 -short -coverprofile=coverage.out ./cogito ./cmd/cogito
 
   test:all:
     desc: Run all the tests (unit + integration). Use this target to get total coverage.
     cmds:
-      - go test -count=1 -coverprofile=coverage.out ./...
+      # FIXME - go test -count=1 -coverprofile=coverage.out ./...
+      - go test -count=1 -short -coverprofile=coverage.out ./cogito ./cmd/cogito
     env: *test-env
 
   test:smoke:
     desc: Simple smoke test of the local executables.
     cmds:
+      - task: build
+      - task: build:copydir
+      - task: test:buildinfo
       - echo '{{.SMOKE_INPUT}}' | ./bin/check
       - echo
       - echo '{{.SMOKE_INPUT}}' | ./bin/in dummy-dir
       - echo
+      - rm -rf /tmp/cogito-test
+      - mkdir -p /tmp/cogito-test
+      - >
+        ./bin/copydir cogito/testdata/one-repo/a-repo /tmp/cogito-test --dot 
+        --template repo_url=https://github.com/foo/bar head=dummyHead
+        branch_name=dummyBranch commit_sha=dummySha
```

Comment:
@marco-m-pix4d, this is actually the line that triggered my comment on renaming `copydir`. I didn't think of it being used as a function in go code where a std lib copy isn't available. Just renaming the binary would be enough, IMO.

Reply:
Ah, good point. Suggestions for the name of this executable? Same as in the previous comment:  `templatedir` or `fleshoutdir`?

Reply:
Yes. I prefer `templatedir`.

Reply:
@odormond 
addressed in last PR of the series.